### PR TITLE
Remove extra keys that are invalid in eslintrc

### LINKED ISSUE #356: The generated eslint config is invalid
I generate the `.eslintrc.js` using the api `neutrino.call('eslintrc')`. Eslint 4.7.2 is installed at project root so that the sublimelinter would use it to do the linting. But it doesn't work right now. I run it manually and found that eslint's config validator is complaining:
```C:\myapp>.\node_modules\.bin\eslint src\store\modules\data.js
ESLint configuration in c:\myapp\.eslintrc.js is invalid:
        - Unexpected top-level property "cwd".

Error: ESLint configuration in c:\myapp\.eslintrc.js is invalid:
        - Unexpected top-level property "cwd".

    at validateConfigSchema (c:\myapp\node_modules\eslint\lib\config\config-validator.js:212:15)
    at Object.validate (c:\myapp\node_modules\eslint\lib\config\config-validator.js:229:5)
    at loadFromDisk (c:\myapp\node_modules\eslint\lib\config\config-file.js:549:19)
    at Object.load (c:\myapp\node_modules\eslint\lib\config\config-file.js:592:20)
    at Config.getLocalConfigHierarchy (c:\myapp\node_modules\eslint\lib\config.js:226:44)
    at Config.getConfigHierarchy (c:\myapp\node_modules\eslint\lib\config.js:180:43)
    at Config.getConfigVector (c:\myapp\node_modules\eslint\lib\config.js:285:21)
    at Config.getConfig (c:\myapp\node_modules\eslint\lib\config.js:328:29)
    at processText (c:\myapp\node_modules\eslint\lib\cli-engine.js:156:33)
    at processFile (c:\myapp\node_modules\eslint\lib\cli-engine.js:216:18)
```

The generated `.eslintrc.js` is
```
module.exports = {
    cwd: 'C:\\myapp',
    useEslintrc: true,
    root: true,
    extensions: ['js', 'jsx'],
    plugins: ['babel', 'vue'],
    extends: ['eslint:recommended', 'prettier'],
    env: { es6: true, browser: true, commonjs: true, node: true },
    parser: 'babel-eslint',
    parserOptions: {
        ecmaVersion: 2017,
        sourceType: 'module',
        ecmaFeatures: {
            objectLiteralDuplicateProperties: false,
            generators: true,
            impliedStrict: true
        }
    },
    settings: {},
    globals: { process: true },
    rules: {
        indent: ['error', 4],
        'linebreak-style': ['off', 'windows'],
        quotes: ['error', 'single'],
        semi: ['error', 'always'],
        'vue/jsx-uses-vars': 2
    },
    overrides: [{ files: '**/**_test.js', rules: { 'no-console': ['off'] } }],
    fix: true
};
```
I found that I have to delete a bunch of top-level properties to give a proper config for eslint to consume:
```
    let config = neutrino.call('eslintrc');
    delete config.cwd;
    delete config.useEslintrc;
    delete config.extensions;
    delete config.fix;
    fs.writeFileSync(
         path.join(__dirname, '.eslintrc.js'),
         'module.exports = ' + JSON.stringify(config)
    );
```

### PR DESCRIPTION
**Pull request by eliperelman**
Fixes #356.

### CODE REVIEW

File: packages/neutrino-middleware-eslint/index.js [hunk 1]
```diff
@@ -15,8 +15,7 @@ const reduceToTrueKeys = reduce((acc, key) => assoc(key, true, acc), {});
 const getEslintRcConfig = pipe(
   getEslintOptions,
   clone,
-  assoc('useEslintrc', true),
-  omit(['failOnError', 'emitWarning', 'emitError']),
+  omit(['failOnError', 'emitWarning', 'emitError', 'cwd', 'useEslintrc', 'fix', 'extensions']),
```
Comment (edmorley) [inline packages/neutrino-middleware-eslint/index.js]: I wonder if we should add a comment explaining why some keys are omitted and linking to:
https://github.com/eslint/eslint/blob/master/conf/config-schema.js
(Used by: https://github.com/eslint/eslint/blob/9d1df92628dd4dd1e70fbb19454008e146387435/lib/config/config-validator.js#L167 )

**Review by edmorley — APPROVED** [review]
Looks good! Perhaps we should add a manual eslint invocation to the Travis run that would catch this? It needn't necessarily lint all the same files but iirc there's a validate config eslint option? (On mobile, so not sure)

File: package.json [hunk 2]
```diff
@@ -51,7 +51,11 @@
   },
   "lint-staged": {
     "*.js": [
-      "packages/neutrino/bin/neutrino lint --fix",
+      "yarn lint --fix",
+      "git add"
+    ],
+    "packages/neutrino/bin/neutrino": [
```
[the remark this answers is not in the document]
Comment (eliperelman) [reply package.json]: Everything in bin is already caught by *.js previously, so I thought this would cover the only other file.

**Comment by eliperelman** [comment]
@edmorley I like that idea! I've implemented something rudimentary that should be usable by Travis.

**Comment by edmorley** [comment]
The Travis tests look great!

Could you also add the new `eslintrc.js` files to those package's `.npmignore`, so they are not bundled?

**Pull request merged**